core/src/main/java/org/mule/runtime/core/internal/event/AbstractEventContext.java

```diff
@@ -386,7 +392,7 @@ public void accept(MonoSink<CoreEvent> sink) {
       if (isResponseDone()) {
         signalPublisherSink(sink);
       } else {
-        synchronized (AbstractEventContext.this) {
+        synchronized (this) {
```

Comment:
you're using the publisher now, is this on purpose?

Reply:
nope. will fix